### PR TITLE
[Migration Required][QA Prep Required] Feature/Fix incorrect last_logged date on existing forks [PLAT-1131]

### PR DESCRIPTION
**Pull request by pattisdr**
❗️ Before merging, make sure this is up-to-date with develop! (is in same release as https://github.com/CenterForOpenScience/osf.io/pull/8808/ - which also has migrations)

## Purpose

Currently, newly created forks receive an incorrect last_logged date. We should find the affected forks and fix the dates.

## Changes

Migration iterates over all forks and find those where the most recent log is the fork log and sets last_logged date to the date of that log

## QA Notes

Create a few forks and note the last_logged date, as well as noting the last logged date of a few nodes.   Make sure the last action that has occurred on the forks was the node_forked action.  Verify that last_logged is the same date as the node_forked action.  If the last log was not related to forks the last_logged date should not have changed.

## Documentation

None needed 

## Ticket

https://openscience.atlassian.net/browse/PLAT-1131

### CODE REVIEW

**Comment by pattisdr** [comment]
@Johnetordoff, that's my understanding, if a fork has had something else changed, you would expect it's last_logged date to be correct already.

File: osf/migrations/0142_auto_20181029_1701.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        node.last_logged = node.logs.order_by('-date')[1].date
+
+    bulk_update(nodes, update_fields=['last_logged'])
+
+def transfer_forked_date(state, schema):
+    """
+    If the most recent node log is forking, transfer that log's date to the node's last_logged field
+    """
+    newest = NodeLog.objects.filter(node=OuterRef('pk')).order_by('-date')
+    nodes = Node.objects.filter(is_fork=True).annotate(latest_log=Subquery(newest.values('action')[:1])).filter(latest_log='node_forked')
```
Comment (sloria) [inline osf/migrations/0142_auto_20181029_1701.py]: I sped up these queries by adding `.filter(is_fork=True)`, which narrows down the result set considerably.

Comment (pattisdr) [reply osf/migrations/0142_auto_20181029_1701.py]: thanks, I forgot about that property

**Comment by sloria** [comment]
Headed to staging3

**Comment by pattisdr** [comment]
Double check if migration needs to be rebased again before merging

**Pull request merged**